drivers/at86rf2xx/at86rf2xx_getset.c

```diff
@@ -130,43 +132,38 @@ static const uint8_t dbm_to_rx_sens[] = { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
                                           0x0d, 0x0e, 0x0e, 0x0e, 0x0f };
 #endif
 
-uint16_t at86rf2xx_get_addr_short(const at86rf2xx_t *dev)
+void at86rf2xx_get_addr_short(const at86rf2xx_t *dev, network_uint16_t *addr)
 {
-    return (dev->netdev.short_addr[0] << 8) | dev->netdev.short_addr[1];
+    memcpy(addr, dev->netdev.short_addr, sizeof(*addr));
```

Comment:
Would you save some bytes of ROM by directly assigning the two bytes here?
Otherwise `memcpy` is more consistent.

Reply:
The nice thing about modern versions of GCC or clang is that they know about most standard C library functions. E.g. if you use `printf("Hello %s\n", "world");`, they will generate a call to `puts("Hello world");` instead.

E.g. when compiling this code:

```C
#include <string.h>

extern int a, b;

void foo(void)
{
	memcpy(&a, &b, sizeof(a));
}
```

with `arm-none-eabi-gcc -std=c99 -Os -Wall -Wextra -pedantic -Werror -S test.c`, the following assembly is generated:

```asm
	.cpu arm7tdmi
	.eabi_attribute 20, 1
	.eabi_attribute 21, 1
	.eabi_attribute 23, 3
	.eabi_attribute 24, 1
	.eabi_attribute 25, 1
	.eabi_attribute 26, 1
	.eabi_attribute 30, 4
	.eabi_attribute 34, 0
	.eabi_attribute 18, 4
	.file	"test.c"
	.text
	.align	2
	.global	foo
	.arch armv4t
	.syntax unified
	.arm
	.fpu softvfp
	.type	foo, %function
foo:
	@ Function supports interworking.
	@ args = 0, pretend = 0, frame = 0
	@ frame_needed = 0, uses_anonymous_args = 0
	@ link register save eliminated.
	ldr	r3, .L2
	ldr	r2, [r3]
	ldr	r3, .L2+4
	str	r2, [r3]
	bx	lr
.L3:
	.align	2
.L2:
	.word	b
	.word	a
	.size	foo, .-foo
	.ident	"GCC: (Alpine Linux) 9.2.0"
```

With modern C compilers it is safe to assume that they only generate the calls to `memcpy()`, if the call is indeed better than generated machine code to do the copying.